### DISCUSSION TITLE
Can I choose wich slices to "remove" using CropOrPad ?

### DISCUSSION THREAD

**Discussion by gkrisp98** [TorchIO-project/torchio#851] [Ideas]
I am working on a project where I have to classify knee MRIs. The original MRIs are of shape (160,384,384). I am using `tio.CropOrPad()` to reduce them. As of my understanding of that transform if I reduce the first dimension some slices will be removed. 
Some slices do not have meaningful information so I think my model should perform better if I remove them.
If I use the transform like this: `tio.CropOrPad((100,224,244))` how do I know which slices are been removed from the object ?
And also, is there any way to choose manually which slices to be left out ?

**Comment by romainVala** [comment]
Hi
with the current implementation you can not really choose which slices, you give a target shape, and torchio compute the need padding or cropping in a symetric way (same number of crop/pad slice up and down)

if you want to change this you may try the `mask_name` option, instead of computing the pad/crop from volume center, it will now compute it from mask center
so if you properly define a mask you may be able to keep only the region of interest
